plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java

```diff
@@ -110,18 +112,38 @@ private T tryConstructor(Constructor<T> constructor, Mapping config, Configurati
 
             for (int i = 0; i < names.length; i++) {
                 final CNode value = config.get(names[i]);
-                if (value == null && parameters[i].getAnnotation(Nonnull.class) != null) {
-                    throw new ConfiguratorException(names[i] + " is required to configure " + target);
-                }
                 final Class t = parameters[i].getType();
+
+                Class<?> clazz = constructor.getDeclaringClass();
+                if (value == null && (parameters[i].isAnnotationPresent(Nonnull.class)   ||
+                    constructor.isAnnotationPresent(ParametersAreNonnullByDefault.class) ||
```

Comment:
May deserve a separate changelog entry. Positively impacts cases not related to collections. May deserve test automation also